lib/logstash/outputs/scalyr.rb

```diff
@@ -561,8 +596,14 @@ def multi_receive(events)
             @logger.warn(message, exc_data)
             exc_commonly_retried = false
           end
-          retry if @running and exc_retries < @max_retries
-          log_retry_failure(multi_event_request, exc_data, exc_retries, exc_sleep)
+
+          @stats_lock.synchronize do
```

Comment:
I guess I might miss something here, but doesn't this statistics update already repeat similar block of code which is located slightly above?

Reply:
Good catch, this is indeed redundant and should not be here.